File: 프로그래머스/lv0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.py
def solution(my_string):
    answer = 0
    
    num = ''
    
    for c in my_string:
        if c.isdigit():
            num += c
        else:
            if len(num) == 0:
                continue
            answer += int(num)
            num = ''
            
    if len(num) != 0:
        answer += int(num)
    
    return answer

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
class Solution {
    public int solution(String str) {
        char[] chrs = str.toCharArray();

        int sum = 0;
        StringBuilder num = new StringBuilder();
        for(int i = 0; i < chrs.length;i++) {
            
            if (check(chrs[i])) {
                num.append(chrs[i]);
            } else if(num.length() > 0) {
                sum += Integer.parseInt(num.toString());
                num = new StringBuilder();
            }
        }
        if(num.length() > 0) {
                sum += Integer.parseInt(num.toString());
                num = new StringBuilder();
        }

        return sum;
    }
    
    static boolean check(char c) {
        return c >= '0' && c <= '9';
    }
}